corpus/00-component-instr-command/src/main.rs

```diff
@@ -0,0 +1,10 @@
+cargo_component_bindings::generate!();
+
+use observe_api::*;
```

Comment:
Would we want to use the `observe_api` functions, or the functions generated by the above `cargo_component_bindings::generate!();` ?

The following seems to work as well:
```rust
cargo_component_bindings::generate!();

use bindings::dylibso::observe::api::*;
// use observe_api::*;

fn main() {
    span_enter("hello world");
    log(LogLevel::Info, b"hello world");
    span_exit();
    println!("Hello, world!");
}
````

Maybe the `observe_api` crate only becomes necessary to import for core modules?

Reply:
Hm! Good point! Using `observe-api` in the component is punning the core module imports – which works right now since they're ABI-compatible, but that might not always be guaranteed to work. (This brings up another issue – `bindings::dylibso::observe::api::LogLevel::Info` and `log::Level::Info` are not represented by the same `u32` number.)

Reply:
What is strange is that the `log` function's signature is different in the generated [WIT](https://github.com/dylibso/observe-sdk/blob/chris/20231003-hack-in-component-support/wit/observe.wit#L17) bindings than it is in the [`observe-api`](https://github.com/dylibso/observe-sdk/blob/chris/20231003-hack-in-component-support/observe-api/rust/src/lib.rs#L17), but with no changes to the host, both components seem to work fine.

Reply:
Ah nevermind, I just expanded the guest bindings and see that the WIT generated log function is `pub fn log(level: LogLevel, msg: &[u8]) {` , your comment on the functions being ABI compatible makes sense now

Reply:
The fact that the log levels are not represented by the same `u32` (as you mentioned) is also troubling. It's probably worth figuring out what strict ABI compatibility between core modules and components would look like. Maybe something we tackle in a second pass.  Perhaps it could be done by using one struct to provide the host implementations for both core modules & components